autonomous/autonomous.py

```diff
@@ -230,59 +199,88 @@ def finished(self) -> None:
         self.intake.deployed = False
 
     def move_next_waypoint(self, cur_time: float) -> None:
-        """Creates the trapazoidal profile to move to the next waypoint"""
-        if self.cur_waypoint >= len(self.waypoints) - 1:
+        """Translates the generated trajectory to move to the next waypoint"""
+        if self.current_movement_idx == len(self.movements) - 1:
+            # last trajectory in the current profile
             self.next_state("finished")
             return
-        # last state in the current profile
-        last_end = self.trap_profile.calculate(self.trap_profile.totalTime())
-        self.cur_waypoint += 1
-        self.trap_profile = self._generate_trap_profile(last_end)
-        self.trap_profile_start_time = cur_time
-
-    def _generate_trap_profile(
-        self, current_state: TrapezoidProfile.State
-    ) -> TrapezoidProfile:
-        """Generates a linear trapazoidal trajectory that goes from current state to goal waypoint"""
-        end_point = trajectory_generator.total_length(
-            self.waypoints_poses[: self.cur_waypoint + 1]
-        )
-        waypoint_type = self.waypoints[self.cur_waypoint].type
-        if (
-            waypoint_type is WaypointType.SHOOT
-            or waypoint_type is WaypointType.PICKUP
-            or self.cur_waypoint >= len(self.waypoints)
-        ):
-            end_speed = 0.0
+
+        self.current_movement_idx += 1
+        self.current_movement = self.movements[self.current_movement_idx]
+        if wpilib.RobotBase.isSimulation():
+            # We aren't actually moving in simulation, so fudge it
+            current_translation = (
+                self.current_movement.trajectory.initialPose().translation()
+            )
         else:
-            end_speed = self.max_speed
-        return TrapezoidProfile(
-            self.linear_constraints,
-            goal=TrapezoidProfile.State(end_point, end_speed),
-            initial=current_state,
+            current_translation = (
+                self.chassis.estimator.getEstimatedPosition().translation()
+            )
+        first_translation = self.current_movement.trajectory.initialPose().translation()
+        # Don't adjust the rotation, just the translation
+        self.current_trajectory = self.movements[
+            self.current_movement_idx
+        ].trajectory.transformBy(
```

Comment:
That was the point of this, and in the pure dead-reckoning case it is redundant. This means that if the vision corrects the position we will always have the origin of the trajectory at our current position on the field.

Reply:
i agree the trajectory should start at the current position but by shift the whole thing wouldn't that also shift the end of the movement aswell, is there a way to set the start of the movement and keep the end where it is?

Reply:
The thinking is that if the odometry is shifted we still want to have the same displacement in the next section as we planned. Shifts are going to be from the odometry being moved by vision, and not from tracking error (because we are going to spend the weekend ensuring there is no tracking error!)

Ultimately, I think we should be looking to get this running well enough that we could do it without vision if we chose.

Reply:
i think not using vision during auto is good idea, if thats the plan should we do that now